lib/internal/http2/compat.js

```diff
@@ -181,6 +181,10 @@ class Http2ServerRequest extends Readable {
     return '2.0';
   }
 
+  get socket() {
+    return this.stream.session.socket;
+  }
```

Comment:
Okay, fair points. 🤔 Will socket be removed from session at some point? I could change the test case to `request.stream.session.socket.alpnProtocol`.